File: typescript-client-sdk/src/sdk/models/operations/getallcrmdeals.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetAllCrmDealsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;
}


export class GetAllCrmDealsResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=deals", elemType: shared.Deal })
  deals?: shared.Deal[];

  @SpeakeasyMetadata({ data: "json, name=nextPageCursor" })
  nextPageCursor?: string;
}


export class GetAllCrmDealsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetAllCrmDealsQueryParams;
}


export class GetAllCrmDealsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetAllCrmDealsResponseBody;
}

File: go-client-sdk/pkg/models/operations/getallcrmdeals.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetAllCrmDealsQueryParams struct {
	AccessToken string   `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool    `queryParam:"style=form,explode=true,name=allFields"`
	Cursor      *string  `queryParam:"style=form,explode=true,name=cursor"`
	Limit       *float64 `queryParam:"style=form,explode=true,name=limit"`
}

type GetAllCrmDealsResponseBody struct {
	Deals          []shared.Deal `json:"deals,omitempty"`
	NextPageCursor *string       `json:"nextPageCursor,omitempty"`
}

type GetAllCrmDealsRequest struct {
	QueryParams GetAllCrmDealsQueryParams
}

type GetAllCrmDealsResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetAllCrmDealsResponseBody
}
